### PR TITLE
Update API doc on performing constrained table queries

### PR DESCRIPTION
**Pull request by QubitPi**
Addresses issue https://github.com/yahoo/fili/issues/560

### CODE REVIEW

File: docs/end-user-api.md [hunk 1]
```diff
@@ -219,6 +219,71 @@ so that we can pull it into Excel and play around with it! No worries, the Fili
...
+          "name":"rowNum",
+          "longName": "rowNum",
+          "uri": "http://localhost:9998/metrics/rowNum"
+        }
+      ]
+    }
+```
+
+Node the line`"availableIntervals": ["2016-05-01 00:00:00.000/2017-05-27 00:00:00.000"]`.
+
```
Comment (michael-mclawhorn) [inline docs/end-user-api.md]: spelling: 'Note'

**Comment by michael-mclawhorn** [comment]
Are the features described here fully implemented in UAD?

**Comment by michael-mclawhorn** [comment]
Is the date time filtering fully implemented?

File: docs/end-user-api.md [hunk 2]
```diff
@@ -219,6 +219,71 @@ so that we can pull it into Excel and play around with it! No worries, the Fili
...
+### Table Availability ###
+Availability for a table(logical table) is defined as the maximal set of intervals that the table may be able to respond
+to. Or put another way, the table can certainly not respond to any intervals beyond the range marked available. 
+
+### Constrained Queries ###
+Constrained availability will be defined as the availability for a table given a set of query constraints. 
+
+Currently the tables/table/timeGrain response lacks an indication of the time range(s) for which the table has data
+(ie. can answer questions), but implementation is going on to expand the Table resource so that it can take pretty much
```
Comment (tarrantzhang) [inline docs/end-user-api.md]: > but implementation is going on to expand the Table resource...

lowercase for `Table`?

File: docs/end-user-api.md [hunk 3]
```diff
@@ -219,6 +219,71 @@ so that we can pull it into Excel and play around with it! No worries, the Fili
...
+the same inputs that the Data resource takes, and would use those inputs to constrain / restrict the available time
```
Comment (tarrantzhang) [inline docs/end-user-api.md]: lowercase for `Data`?

**Comment by QubitPi** [comment]
@michael-mclawhorn I'm happy to answer "Is the date time filtering fully implemented?":

I think it has been implemented - https://github.com/yahoo/fili/blob/master/fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/TablesServlet.java#L581

**Comment by QubitPi** [comment]
I'll double check code before merging this

**Pull request merged**